lib/config/rules/functionality.js

```diff
@@ -0,0 +1,80 @@
+// Functionality Specific:
+// These rules catch common errors in JS programming or
+// otherwise confusing constructs that are prone to producing bugs:
+// Reference https://palantir.github.io/tslint/rules/
+
+module.exports = {
+  // Bans the use of specific functions or global methods.
+  ban: false,
+  // Enforces braces for `if`/`for`/`do`/`while` statements.
+  curly: true,
+  // Requires a `for ... in` statement to be filtered with an `if` statement.
+  forin: true,
+  // Disallows importing the specified modules directly via `import` and `require`.
+  // Instead only sub modules may be imported from that module.
+  'import-blacklist': false,
+  // Only allows labels in sensible locations.
+  'label-position': true,
+  // Disallows use of `arguments.callee`. <FEEDBACK>
+  'no-arg': false,
+  // Disallows bitwise operators.
+  'no-bitwise': false,
+  // Disallows any type of assignment in conditionals.
+  'no-conditional-assignment': true,
+  // Bans the use of specified `console` methods.
+  'no-console': [true, 'log', 'error'],
+  // Disallows access to the constructors of `String`, `Number`, and `Boolean`. <FEEDBACK>
+  'no-construct': false,
+  // Disallows `debugger` statements.
+  'no-debugger': true,
+  // Disallows duplicate variable declarations in the same block scope.
+  'no-duplicate-variable': true,
+  // Disallows empty blocks.
+  'no-empty': true,
+  // Disallows `eval` function invocations.
+  'no-eval': true,
+  // Disallows iterating over an array with a for-in loop. <FEEDBACK>
+  'no-for-in-array': false,
+  // Disallow type inference of {} (empty object type) at function and constructor call sites <FEEDBACK>
+  'no-inferred-empty-object-type': false,
```

Comment:
I believe this is what happens. If the type of a value is not explicitly specified the resulting inference is the empty object type, {}
See the [Best common type](https://www.typescriptlang.org/docs/handbook/type-inference.html) section